test/suite/test_eviction02.py

```diff
@@ -26,89 +26,77 @@
-
-    def get_stat(self, stat, uri = ""):
-        stat_cursor = self.session.open_cursor(f'statistics:{uri}')
-        val = stat_cursor[stat][2]
-        stat_cursor.close()
-        return val
-
-    def populate(self, uri, start_key, num_keys, value):
-        c = self.session.open_cursor(uri, None)
-        for k in range(start_key, num_keys):
-            self.session.begin_transaction()
-            c[k] = value
-            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(k + 1))
-        c.close()
-
-    def evict_cursor(self, uri, nrows):
-        # Configure debug behavior on a cursor to evict the page positioned on when the reset API is used.
-        evict_cursor = self.session.open_cursor(uri, None, "debug=(release_evict)")
-        self.session.begin_transaction("ignore_prepare=true")
-        for i in range (1, nrows + 1):
-            evict_cursor.set_key(i)
-            evict_cursor.search()
-            if i % 10 == 0:
-                evict_cursor.reset()
-        evict_cursor.close()
-        self.session.rollback_transaction()
-
     def test_evict(self):
         create_params = 'key_format=i,value_format=S'
         nrows = 10000
+        prev_obsolete_tw_value = 0
+        # Stats may have a stale value, allow some buffer.
+        threshold = self.obsolete_tw_max * 1.5
         uri = 'table:eviction02'
         value = 'k' * 1024
 
         self.session.create(uri, create_params)
 
-        prev_obsolete_tw_value = 0
         for i in range(10):
-            # Append some data.
-            self.populate(uri, nrows * i, nrows * (i + 1), value)
+            # Add some data.
+            start_key = nrows * i
+            num_keys = nrows * (i + 1)
+            self.populate(uri, start_key, num_keys, value)
 
-            # Checkpoint with cleanup.
-            self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(nrows * (i + 1)))
+            # Make all the inserted data stable and checkpoint to make everything clean.
+            self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(num_keys))
             self.session.checkpoint()
 
-            # Check statistics.
-            current_obsolete_tw_value = self.get_stat(stat.dsrc.cache_eviction_dirty_obsolete_tw, uri)
-            if i > 0:
-                self.check_stat(prev_obsolete_tw_value, current_obsolete_tw_value)
-            prev_obsolete_tw_value = current_obsolete_tw_value
+            # Bump the oldest timestamp to make some of the previously inserted data globally
+            # visible. This makes any time window informaton associated with that data obsolete and
+            # eligible for cleanup.
+            self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(num_keys // 2))
 
-            self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(nrows * (i + 1)))
+            # Eviction should perform clean eviction here which triggers the assessement of obsolete
```

Comment:
This is not true without the debug option. We will evict the page without reviewing the time window information.

Reply:
Removed that function in 9a20093a4914788383fbc5011a4cbfd5e6448c5e